components/clock.rst

```diff
@@ -0,0 +1,95 @@
+.. index::
+   single: Clock
+   single: Components; Clock
+
+.. _`clock-component`:
+
+The Clock Component
+===================
+
+.. versionadded:: 6.2
+
+    The Clock component was introduced in Symfony 6.2
+
+The Clock component decouples applications from the system clock. This improves
+testability of time-sensitive logic.
+
+The component provides a ``ClockInterface`` with the following implementations
+for different use cases:
+
+* ``NativeClock`` provides a way to interact with the system clock;
+* ``MockClock`` is commonly used in tests as a replacement for the
+  ``NativeClock`` to be able to freeze and change the current time using either
+  ``sleep()`` or ``modify()``;
+* ``MonotonicClock`` relies on ``hrtime()`` and provides a high resolution,
+  monotonic clock, when you need a precise stopwatch.
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/clock
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+A clock service replaces creating a new ``DateTime`` or
+``DateTimeImmutable`` object for the current time. Instead, you inject the
+``ClockInterface`` and call ``now()``. By default, your application will likely
+use a ``NativeClock``, which always returns the current system time. In tests it is replaced with a ``MockClock``. The
+The ``MockClock`` is instantiated with a time and does not move forward on its own. The time is
+fixed until ``sleep()`` or ``modify()`` are called. This gives you full control over what your code
+assumes is the current time.
```

Comment:
I think we should move the mock clock information the testing section below (and rename it to "Testing using MockClock"?). Then we can also explain that sleep doesn't actually sleep using the mock clock.